charm_test.go

```diff
@@ -12,19 +12,41 @@ import (
 	"path/filepath"
 	stdtesting "testing"
 
+	"github.com/juju/testing"
 	jc "github.com/juju/testing/checkers"
 	"github.com/juju/utils/fs"
 	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/charm.v6"
 	"gopkg.in/yaml.v2"
 
-	"gopkg.in/juju/charm.v6"
+	coretesting "github.com/juju/juju/testing"
 )
 
 func Test(t *stdtesting.T) {
 	gc.TestingT(t)
 }
 
-type CharmSuite struct{}
+type CharmSuite struct {
+	coretesting.BaseSuite
+}
+
+var expectedArgs = struct {
```

Comment:
don't need this struct as there are separate tests.
just use the args slices directly in the relevant tests

Reply:
done